src/audio/pipeline.c

```diff
@@ -616,6 +616,30 @@ static void pipeline_comp_trigger_sched_comp(struct pipeline *p,
 	list_item_append(&p->list, &ctx->pipelines);
 }
 
+/* if capture START/RELEASE trigger originated on dailess pipeline and reached
+ * inactive pipeline as it's source, then we indicate that it's blocked.
+ */
+static inline int
+pipeline_depender_is_blocked(struct comp_dev *dependee,
+			     struct pipeline_walk_context *ctx,
+			     int dir)
+{
+	struct pipeline_data *ppl_data = ctx->comp_data;
+	int is_trigger_start_or_release = ppl_data->cmd == COMP_TRIGGER_START ||
+					  ppl_data->cmd == COMP_TRIGGER_RELEASE;
+	int is_capture = dir == SOF_IPC_STREAM_CAPTURE;
+	struct comp_dev *pipe_source = ppl_data->start->pipeline->source_comp;
+	int has_dai_source = pipe_source &&
+			     dev_comp_type(pipe_source) == SOF_COMP_DAI;
+	int is_dependee_active = dependee->state == COMP_STATE_ACTIVE;
```

Comment:
What is dependee ? It it source or sink component ?

Reply:
dependee is source component from the other pipeline, on which triggered pipeline depends.

Reply:
Then we can call it rsrc. (remote src) .